auth/src/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -237,30 +237,37 @@ export class VSCodeAzureSubscriptionProvider extends vscode.Disposable implement
      *
      * @returns A client, the credential used by the client, and the authentication function
      */
-    private async getSubscriptionClient(tenantId?: string, scopes?: string[]): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
-        const armSubs = await import('@azure/arm-resources-subscriptions');
-        const session = await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true });
-        if (!session) {
+    private async getSubscriptionClient(tenantId?: string): Promise<{ client: SubscriptionClient, credential: TokenCredential, authentication: AzureAuthentication }> {
+        const initialSession = await getSessionFromVSCode(undefined, tenantId, { createIfNone: false, silent: true });

```

Comment:
Can you not call `getSessionFromVSCode` here and just have the scopes be a parameter that gets passed in? That way the async work will be done at this point.

Reply:
Should the MS auth provider be used directly instead of trying to make this subscription-listing package do it?

Reply:
I tested this locally and it works. I think it would work for Jason's use case and also keep things working on our end. However, I don't love it because any of our existing requests that pass scopes here would now be async. I'm not sure if that's an issue, but it could be since it was an issue for the zip deploy requests.
```ts
            authentication: {
                getSession: async (scopes) => {
                    if (scopes) {
                        return await getSessionFromVSCode(scopes, tenantId, { createIfNone: false, silent: true });
                    } else {
                        return initialSession;
                    }
                }
            }
```

Reply:
The SDKs aren't using `getSession`, that's just meant to mirror the VSCode API for getting auth sessions. The SDKs will use `getToken`. In fact I don't think anything should be directly using `getSession`.

Reply:
A fix to the deployment issue has been merged.

Reply:
Great find @JasonYeMSFT!